File: fp51.py
t = int(input())
while t > 0:
    t-=1 
    n = int(input())
    x = list(map(int,input().split()))
    max = -1
    min = 11
    z = 0
    for i in range(1,n):
        k = x[i] - x[i-1]
        if k <= 2 :
            z += 1
        else:
            if z <= min :
                min = z 
            if z >= max :
                max = z  
            z = 0
    if z <= min :
        min = z 
    if z >= max :
        max = z 
    print(min+1,max+1) 

File: fp51.cpp
// codecheaf 3 may 20 b
#include<bits/stdc++.h>
using namespace std;

int main()
{
    int t;
    cin>>t;
    while(t--)
    {
        long long x,y,z,l,r;
        cin>>x>>y>>l>>r;
        z=x|y;
        int h=min(x,y);
        if(h==0)
        cout<<0<<endl;
        else if(z<=r)
        cout<<z<<endl;
        else
        {
            h=max(x,y);
            int s=log(h)+1;
            long long  p=powl(2,s)-1;
            if(((p|x) && (p|y) )||((r|x) && (r|y)))
                cout<<0<<endl;
            else
                cout<<p<<endl;
        }
    }
    return 0;
}